Sources/Datadog/Core/DD/CoreLogger.swift

```diff
@@ -0,0 +1,98 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+
+/// The `CoreLogger` provides methods to collect debug information and detect execution errors of the Datadog SDK.
+internal struct CoreLogger: CoreLoggerType {
+    /// The prefix applied to all core logs.
+    private static let prefix = "[DATADOG SDK] 🐶 → "
+
+    /// The date provider for annotating core logs.
+    private let dateProvider: DateProvider
+    /// Formatter used to format the time accordingly for local device.
+    private let dateFormatter: DateFormatterType
+    /// The print function.
+    private let printFunction: (String) -> Void
```

Comment:
Haven't ever used [`TextOutputStream`](https://developer.apple.com/documentation/swift/textoutputstream). Let me look into this 👀🙂!

Reply:
@maxep I'm not sure on the actual benefit coming from `TextOutputStream`. The way proposed in above snippet ☝️ would mix formatting and displaying the string across two units (emoji + `DDError` in logger and prefix + time in the stream). If we want to enable cross-platform SDKs to customize DD logs, then IMO we could expose `format` property on `DD.logger` (e.g. allow custom prefix). The stream would come powerful if we want to redirect the actual output from STDOUT to sth else (file or STDERR) so we can look into this if we need it. Do I miss something?

Reply:
Yes indeed, the idea was to delegate the formatting and printing to a single object so crossplatforms could provide their own. But formatting depending in the entry level won't fit in `TextOutputStream`, we could have `level` property in it but it won't look nice.

Now, with this new interface, crossplatform could provide their own implementation of a `CoreLogger`, so we are good 👍  Wouldn't be easier to only require a single method? like:
```swift
protocol CoreLogger {
  func log(_ level: LogLevel, message: @autoclosure () -> String)
}
```
All other methods can be defined in the `extension`, WDYT? It would make it easier to comply to!